src/main.rs

```diff
@@ -1,10 +1,72 @@
 use smbioslib::*;
 
+
+fn print_usage(program: &str, opts: getopts::Options) {
+	let brief = format!("Usage: {} [-f|-o FILE]", program);
+	print!("{}", opts.usage(&brief));
+}
+
 fn main() {
-    match table_load_from_device() {
-        Ok(table) => {
-            println!("table_data: {:#?}", table);
+    let file_option = "f";
+    let output_option = "o";
+
+    let args: Vec<String> = std::env::args().collect();
+    let mut opts = getopts::Options::new();
+    opts.optopt(file_option, "", "read smbios table from file", "FILE");
+    opts.optopt(output_option, "", "dump smbios table to a file", "FILE");
+
+    let matches = match opts.parse(&args[1..]) {
+        Ok(m) => { m },
```

Comment:
We can use [this](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html) 2018 rust design for main.

Like in our lib.rs we can add some code constraints:
```rust
#![warn(missing_docs)]
#![deny(rust_2018_idioms)]
```

Reply:
Great suggestion, it simplified a lot.